.github/workflows/push-image.yaml

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright 2022 The OpenFunction Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: WorkFlow for Building image
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+    paths:
+      - '.github/workflows/**'
+      - 'pkg/**'
+      - 'test/**'
+      - 'Dockerfile'
+      - 'go.mod'
+      - 'main.go'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    name: Build Image
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.x
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
```

Comment:
try this https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fb-image.yaml#L40 ?

Reply:
ok, i'll try